pyproject.toml

```diff
@@ -16,12 +16,9 @@ dependencies = [
   "numba",
   "numpy>=1.21",
   "pandas",
-  "pyarrow",
-  # TODO: once we no longer support Python 3.7, remove this and pin to pyarrow >= 14.0.1
```

Comment:
Following on from my review comment: let's keep `pyarrow-hotfix` _and_ update the reason-why-it-is-needed comment (which has gone stale)

Reply:
pyarrow-hotfix has been restored.